app/js/components/common/notification/notification.component.js

```diff
@@ -0,0 +1,43 @@
+const template = require("./notification.tpl.html");
+
+const NotificationComponent = {
+  template,
+  controller: [
+    "$log",
+    "$rootScope",
+    "$scope",
+    "$timeout",
+    function($log, $rootScope, $scope, $timeout) {
+      "ngInject";
+      $log = $log.getInstance("NotificationComponent", true);
+      $log.log("");
+
+      let vm = this;
+
+      vm.removeNotification = () => {
+        $log.log(`removed`);
+        vm.message = "";
+        vm.type = "";
+      };
+
+      $rootScope.$on("TRIGGERED_NOTIFICATION", (evt, data) => {
+        const { message, type } = data;
+        $log.log(`triggered: ${type} message=${message}`);
+
+        vm.message = message;
+        vm.type = type;
+
+        $scope.$apply();
```

Comment:
This might be needed because you're wiring up your own notification system that does things outside of Angular's digest cycle. That makes it sound like I know what I'm talking about regarding inner workings of Angular, but that's not really the case, so consider this just a hunch.